File: olzdata/public/js/admin_clawback.js
(function ($, api_url, Vue, swal, axios, location, moment, undefined) {

    const client = commissionEngine.createAccessClient();
    commissionEngine.setupAccessTokenJQueryAjax();

    $.fn.ddatepicker = $.fn.datepicker; // jquery-ui is overriding the bootstrap-datepicker

    let $dtLogs;
    let $dt;

    const vm = new Vue({
        el: "#app-clawback",
        data: function () {
            return {

                autocompleteUrl: `${api_url}common/autocomplete/members`,

                filters: {
                    startDate: null,
                    endDate: null,
                    memberId: null,
                },
                products: [],
                transaction_id: null,
                is_clawback: 0,

                order: {
                    order_id: null,
                    purchaser_id: null,
                    purchaser: null,
                    sponsor_id: null,
                    sponsor: null,
                    sub_total: null,
                    transaction_date: null,
                    tax: null,
                    shipping_fee: null,
                    total: null,
                    percentage_off: null,
                    amount_off: null,
                    commission_value: null,
                    is_full_order: null,
                    new_purchaser_id: null,
                    is_clawback: 0
                },

                error: {
                    message : null,
                },

                is_saving: 0,

            };
        },
        mounted() {
            this.initializeDataTables();
            this.initializeDatePicker();
            this.initializeJQueryEvents();
        },
        methods: {
            initializeDataTables() {
                let _this = this;

                $dt = $("#table-orders").DataTable({
                    processing: true,
                    serverSide: true,
                    responsive: true,
                    pageLength: 25,
                    ajax: {
                        "url": api_url + 'admin/clawback/',
                        "data": function (d) {
                            d.startDate = $("#start-date").val();
                            d.endDate = $("#end-date").val();
                            d.memberId = _this.filters.memberId;
                        },
                    },
                    columns: [
                        {data: 'order_id'},
                        {data: 'invoice'},
                        {data: 'purchaser', className: 'text-left text-capitalize'},
                        {data: 'description', className: 'text-left text-capitalize'},
                        {data: 'transaction_date'},
                        {data: 'commission_value'},
                        {data: 'amount_paid'},
                        {data: 'set_by', className: 'text-left'},
                        {
                            data: 'action',
                            width: '200px',
                            render: function (data, type, row, meta) {
                                return '<div class="btn-group-xs" role="group" aria-label="...">' +
                                    '<button ' + ((+row.is_clawback && +row.is_per_product) ? 'disabled' : '') + ' type="button" class="btn btn-danger btn-order-refund" style="margin-right: 5px;">Refund</button>' +
                                    '<button ' + ((+row.is_clawback && !+row.is_per_product) ? 'disabled' : '') + ' type="button" class="btn btn-info btn-view-items" style="margin-right: 5px;">Refund By Items</button>' +
                                    '<button type="button" class="btn btn-warning btn-order-move">Move</button>' +
                                    '</div>';
                            }
                        },
                    ],
                    columnDefs: [

                        {responsivePriority: 1, targets: 0},
                        {responsivePriority: 2, targets: -1},
                    ]
                });

                $dtLogs = $("#table-logs").DataTable({
                    processing: true,
                    serverSide: true,
                    ajax: {
                        url : api_url + 'admin/move-order/logs',
                        data: function(d) {

                        },
                    },
                    order: [[ 5, 'desc' ]],
                    columns     : [
                        {data    : 'order_id'},
                        // {data    : 'new_purchaser'},
                        {
                            data: 'new_purchaser',
                            render: function ( data, type, row, meta ) {

                                if(row.new_user_id == row.old_user_id)
                                {
                                    return '<span class="label label-info">No changes</span>';
                                }

                                return data;
                            }
                        },
                        // {data    : 'new_sponsor'},
                        {data    : 'old_purchaser'},
                        // {data    : 'old_sponsor'},
                        // {data    : 'new_transaction_date'},
                        {
                            data: 'new_transaction_date',
                            render: function ( data, type, row, meta ) {

                                if(row.old_transaction_date == row.new_transaction_date)
                                {
                                    return '<span class="label label-info">No changes</span>';
                                }

                                return data;
                            }
                        },
                        {data    : 'old_transaction_date'},
                        {data    : 'created_at'},
                        {data    : 'changed_by'},
                    ]
                });
            },
            initializeDatePicker() {
                let _this = this;

                $('#start-date').ddatepicker({
                    "setDate": new Date(),
                    "format": "yyyy-mm-dd"
                }).on('changeDate', function (e) {
                    $('#end-date').ddatepicker('setStartDate', e.date);

                    if ($('#end-date').ddatepicker('getDate') < e.date) {
                        $('#end-date').ddatepicker('setDate', e.date);
                    }
                });

                $('#end-date').ddatepicker({
                    "setDate": new Date(),
                    "startDate": new Date(),
                    "format": "yyyy-mm-dd"
                });

                $('#start-date').ddatepicker('setDate', new Date());
                $('#end-date').ddatepicker('setDate', new Date());

                // Transaction Date
                $('#transaction-date').datetimepicker({
                    defaultDate: moment(),
                    format: 'YYYY-MM-DD HH:mm:ss',
                });

                $('#transaction-date').on('dp.change', e => {_this.order.transaction_date = $('#transaction-date').val()});
            },
            initializeJQueryEvents() {

                let _this = this;

                $('#button-view-orders').on('click', function () {
                    $dt.clear().draw();
                    $dt.responsive.recalc();
                });

                $('#table-orders tbody').on('click', '.btn-order-refund', function () {
                    let data = $dt.row($(this).parents('tr')).data();
                    _this.showOrder(data);
                });

                $('#table-orders tbody').on('click', '.btn-view-items', function () {
                    let data = $dt.row($(this).parents('tr')).data();
                    _this.getProducts(data.order_id, data.is_clawback, data);
                });

                $('#table-orders tbody').on('click', '.btn-order-move', function () {
                    let data = $dt.row($(this).parents('tr')).data();
                    _this.showMoveOrder(data);
                });
            },
            readonlyProduct: function(order) {
                return +order.is_clawback && !+order.is_per_product;
            },
            formatMoney: function(value) {
                if(value == null || value == undefined) return 0.00;

                return parseFloat(value).toFixed(2);
            },
            changeIsFullOrder: function(event) {
                if(this.order.is_full_order) {
                    this.order.amount_off = 0;
                    this.order.percentage_off = 0;
                }
            },
            keyupAmount: function(event) {
                this.order.percentage_off = null;
            },
            keyupPercent: function(event) {
                this.order.amount_off = null;
            },
            showOrder: function(data) {
                this.error.message = null;

                this.order = JSON.parse(JSON.stringify(data));
                this.order.amount_off = data.amount_off ? data.amount_off : data.commission_value;
                // this.order.is_full_order = data.is_full_order;
                $('#modal-refund-order').modal('show');
            },
            getProducts: function (transaction_id, is_clawback, order) {

                this.order = JSON.parse(JSON.stringify(order));
                this.transaction_id = transaction_id;

                client.get(`admin/clawback/order-products/${transaction_id}`).then(response => {
                    this.products = response.data;
                    this.is_clawback = is_clawback;
                    this.error.message = null;
                    $('#modal-order-items').modal('show');
                }).catch(error => {
                    swal('Unable to fetch!','','error');
                });
            },

            p: function (index, name) {
                return 'products[' + index + '].' + name;
            },
            onSubmit: function (type) {

                // if(this.is_clawback) return;

                let _this = this;
                swal({
                    title: "Are you sure you want to clawback/refund Order ID " + this.transaction_id + " - Items? " + (type == 'commission' ? "(Commission Only)" :"(Merchant & Commission)"),
                    text: "You cannot undo this.",
                    type: "warning",
                    showCancelButton: true,
                    confirmButtonClass: "btn-danger",
                    confirmButtonText: "Confirm Refund",
                    closeOnConfirm: false,
                    showLoaderOnConfirm: true,
                }, function () {

                    client.post(`admin/clawback/refund-order-products`, {
                        products : _this.products,
                        transaction_id: _this.transaction_id,
                        set_user_id: $('#member').val(),
                        type: type
                    }).then(response => {

                        $dt.clear().draw();
                        $dt.responsive.recalc();

                        _this.products = [];
                        _this.transaction_id = null;

                        $('#modal-order-items').modal('hide');
                        swal("Successfully added for clawback/refund!", "", "success");
                    }).catch(this.axiosErrorHandler).finally(()=> {

                    });

                });
            },
            onSubmitRefund: function (type){

                if(this.order.is_clawback) return;

                var $this = this;
                let $form = $('#form-refund-order');

                swal({
                        title: "Are you sure you want to clawback/refund Order ID " + $('#form-refund-order [name="transaction_id"]').val() + "? " + (type == 'commission' ? "(Commission Only)" :"(Merchant & Commission)"),
                        text: "You cannot undo this.",
                        type: "warning",
                        showCancelButton: true,
                        confirmButtonClass: "btn-danger",
                        confirmButtonText: "Confirm Refund",
                        closeOnConfirm: false,
                        showLoaderOnConfirm: true,
                    },
                    function () {

                        // if (!proceed) {
                        //     swal.close();
                        //     return;
                        // }
                        console.log($form.serialize() + "&set_user_id=" + $('#member').val() + "&type=" + type);
                        $.post(api_url + 'admin/clawback/refund-order', $form.serialize() + "&set_user_id=" + $('#member').val() + "&type=" + type, function (result) {
                            console.log(result);
                            $('#modal-refund-order').modal('hide');
                            $dt.clear().draw();
                            $dt.responsive.recalc()
                            $this.error.message = null;
                            $this.order = {
                                order_id: null,
                                purchaser: null,
                                sub_total: null,
                                tax: null,
                                shipping_fee: null,
                                total: null,
                                percentage_off: null,
                                amount_off: null,
                                commission_value: null,
                                is_full_order: null,
                                is_clawback: 0
                            };
                            swal("Successfully added for clawback/refund!", "", "success");
                        }, 'json').fail(function (xhr, status, error) {
                            if(xhr.responseJSON.message != undefined && xhr.responseJSON.message == "Validation error") {
                                let errors = xhr.responseJSON.errors;
                                $this.error.message = errors[Object.keys(errors)[0]][0];
                            } else if(xhr.responseJSON.error != undefined && typeof xhr.responseJSON.error === "string") {
                                $this.error.message = xhr.responseJSON.error;
                            } else if(xhr.responseJSON.message != undefined) {
                                $this.error.message = xhr.responseJSON.message;
                            } else {
                                $this.error.message = 'Something went wrong!';
                            }
                            swal.close();
                        });

                    });
            },
            showMoveOrder(order) {
                this.order.invoice = order.invoice;
                this.order.order_id = order.order_id;
                this.order.purchaser = order.purchaser;
                this.order.purchaser_id = order.purchaser_id;
                this.order.sponsor = order.sponsor;
                this.order.sponsor_id = order.sponsor_id;
                this.order.transaction_date = order.transaction_date;
                this.order.new_purchaser_id = null;
                this.error.message = null;

                $('#modal-move').modal({backdrop: 'static', keyboard: false});
            },
            save() {
                swal({
                    title: `Are you sure you want to update Order ID ${this.order.order_id}?`,
                    // text: "You cannot undo this.",
                    type: "warning",
                    confirmButtonClass: "btn-success",
                    confirmButtonText: "Confirm",
                    cancelButtonText: "Cancel",
                    showCancelButton: true,
                    closeOnConfirm: false,
                    showLoaderOnConfirm: true,
                }, function () {

                    this.is_saving = 1;

                    client.post(`admin/move-order/orders/${this.order.order_id}/change`, {modified: $('#member').val(), ...this.order}).then(response => {

                        $('#modal-move').modal('hide');
                        // swal.close();
                        swal("Successfully updated!", "", "success");
                        $dt.clear().draw();
                        $dt.responsive.recalc();
                        $dtLogs.clear().draw();
                        $dtLogs.responsive.recalc();

                    }).catch(this.axiosErrorHandler).finally(()=> {

                        this.is_saving = 0;
                    });

                });
            },
            axiosErrorHandler(error) {

                let data = commissionEngine.parseAxiosErrorData(error.response.data);

                this.error.message = data.message;
                this.error.type = data.type;
                this.error.data = data.data;

                swal(this.error.message, "", "error");
            },
        },
        computed: {
            displayPurchaser() {
                return this.order.purchaser_id + ": " + this.order.purchaser;
            },
            displaySponsor() {
                return this.order.sponsor_id + ": " + this.order.sponsor;
            }
        }
    });


}(jQuery, window.commissionEngine.API_URL, Vue, swal, axios, window.location, moment));

File: 515data/pages/admin_clawback.pl
print <<EOS;

<link rel="stylesheet" href="$commission_engine_api_url/css/app.css?v=$app_css_version" />
<link rel="stylesheet" href="https://cdn.datatables.net/responsive/2.2.3/css/responsive.bootstrap.min.css">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/eonasdan-bootstrap-datetimepicker/4.17.47/css/bootstrap-datetimepicker.min.css" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/select2/4.0.7/css/select2.min.css" />
<link rel="stylesheet" href="$commission_engine_api_url/css/datepicker.css?v=$app_css_version" />
<link rel="stylesheet" href="$commission_engine_api_url/css/select2-bootstrap.css"/>
<link rel="stylesheet" href="$commission_engine_api_url/css/admin_clawback.css?v=1" />

<div class="tool-container tool-container--default" id="app-clawback">

    <div class="row">
        <div class="col-md-12">
            <h4>CLAWBACK</h4>
            <hr />
        </div>
    </div>
    <div class="row">
        <div class="col-lg-12" >
            <ul class="nav nav-tabs" role="tablist" id='nav-tab-report'>
                <li role="presentation" class="active"><a href="#tab-orders" aria-controls="tab-orders" role="tab" data-toggle="tab">Orders</a></li>
                <!--<li role="presentation"><a href="#tab-logs" aria-controls="tab-logs" role="tab" data-toggle="tab">Move Logs</a></li>-->
            </ul>
            <!-- Tab panes -->
            <div class="tab-content" style="padding: 15px;border: 1px solid #ddd;border-top: none;">
                <div role="tabpanel" class="tab-pane active" id="tab-orders">

                    <form class="form-horizontal">
                        <div class="form-group">
                            <div class="col-sm-3 col-md-3 col-lg-3">
                                <label for="start-date">From</label>
                                <input id="start-date" type="text" class="form-control flat">
                            </div>
                            <div class="col-sm-3 col-md-3 col-lg-3">
                                <label for="end-date">To</label>
                                <input id="end-date" type="text" class="form-control flat">
                            </div>
                        </div>
                        <div class="form-group">
                            <div class="col-sm-3 col-md-3 col-lg-3">
                                <label for="purchaser-id">Associate (Optional)</label>
                                <select2-autocomplete-member id="purchaser-id" :url="autocompleteUrl" v-model="filters.memberId"></select2-autocomplete-member>
                            </div>
                            <div class="col-sm-3 col-md-3 col-lg-3">
                                <label for="purchaser-id">POD/PDR (Optional)</label>
                                <select2-autocomplete-pod id="pod-id" :url="autocompletePodUrl" v-model="filters.pod"></select2-autocomplete-pod>
                            </div>
                        </div>
                        <div class="form-group">
                            <div class="col-sm-12">
                                <button id="button-view-orders" type="button" class="btn btn-primary" >View</button>
                            </div>
                        </div>
                    </form>
                    <div class="table-responsive">
                        <table id="table-orders" class="table table-striped table-bordered dt-responsive nowrap" style="width:100%;">
                            <thead class="table__header table__header--bg-primary">
                            <tr class="table__row">
                                <th class="table__cell">POD/PDR#</th>
                                <th class="table__cell">Associate ID</th>
                                <th class="table__cell">Associate Name</th>
                                <th class="table__cell">Customer</th>
                                <th class="table__cell">Account Type</th>
                                <th class="table__cell">Date Accepted</th>
                                <th class="table__cell">Date Started Flowing</th>
                                <th class="table__cell">Status</th>
                                <th class="table__cell">Action</th>
                            </tr>
                            </thead>
                            <tbody class="table__body">
                            </tbody>

                        </table>
                    </div>
                </div>
                <div role="tabpanel" class="tab-pane" id="tab-logs">
                    <div class="table-responsive">
                        <table id="table-logs" class="table table-bordered" style="width: 100%;">
                            <thead class="table__header table__header--bg-primary">
                            <tr class="table__row">
                                <th class="table__cell">Order ID</th>
                                <th class="table__cell">New Purchaser</th>
                                <th class="table__cell">Old Purchaser</th>
                                <th class="table__cell">New Transaction Date</th>
                                <th class="table__cell">Old Transaction Date</th>
                                <th class="table__cell">Date</th>
                                <th class="table__cell">Changed by</th>
                            </tr>
                            </thead>
                            <tbody class="table__body">
                            </tbody>
                        </table>
                    </div>

                </div>
            </div>

        </div>
    </div>

    <div class="modal fade" id="modal-order-items" role="dialog" aria-labelledby="modal-order-items-label">
        <div class="modal-dialog modal-xl modal-lg" role="document">
            <form class="modal-content" v-on:submit.prevent>
                <div class="modal-header">
                    <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
                    <h4 class="modal-title" id="modal-order-items-label">ORDER ID {{ transaction_id }} - ORDER ITEMS</h4>
                </div>
                <div class="modal-body">

                    <div class="table-responsive">
                        <table id="table-order-items" class="table table table-striped" >
                            <!--Table head-->
                            <thead class="table__header table__header--bg-primary">
                            <tr class="table__row">
                                <th class="table__cell col-md-2">ID</th>
                                <th class="table__cell col-md-3 text-left">Name</th>
                                <th class="table__cell col-md-1">Qty</th>
                                <th class="table__cell col-md-1">CV</th>
                                <th class="table__cell col-md-1">Total CV</th>
                                <th class="table__cell col-md-1">Refunded Qty</th>
                                <th class="table__cell col-md-2">Total CV Refunded</th>
                                <th class="table__cell col-md-1">Refund Qty</th>
                            </tr>
                            </thead>
                            <!--Table head-->
                            <tbody class="table__body">
                            <tr v-if="products.length == 0" class="table__row text-center">
                                <td colspan="6">Fetching <i class="fa fa-spinner fa-pulse"></i></td>
                            </tr>
                            <tr v-for="(product, index) in products" class="table__row">
                                <td class="table__cell text-center">{{ product.transaction_product_id }}</td>
                                <td class="table__cell text-left">
                                    {{ product.name }}
                                    <ul  style="padding-left: 20px;font-size: 12px;">
                                        <li class="text-danger" v-for="(option, i) in product.options">{{ option.name }}: {{ option.value }}</li>
                                    </ul>
                                </td>
                                <td class="table__cell text-center">{{ product.quantity }}</td>
                                <td class="table__cell text-right">{{ product.price }}</td>
                                <td class="table__cell text-right">{{ product.total }}</td>
                                <td class="table__cell text-center">{{ product.refunded_quantity }}</td>
                                <td class="table__cell text-right">{{ product.refunded_amount }}</td>
                                <td class="table__cell text-center">
                                    <input :name="p(product.transaction_product_id, 'order_id')" v-model="product.order_id" type="hidden">
                                    <input :name="p(product.transaction_product_id, 'transaction_product_id')" v-model="product.transaction_product_id" type="hidden">
                                    <input :readonly="readonlyProduct(order) == 1 || product.quantity < 1" :name="p(product.transaction_product_id, 'refund_quantity')" v-model="product.refund_quantity" type="text" class="form-control input-sm">
                                </td>
                            </tr>
                            </tbody>
                        </table>
                    </div>
                    <div id="form-refund-order-products-error" class="form-group" v-show="error.message != null">
                        <div class="col-sm-12 has-error text-center">
                            <span class="help-block">{{ error.message }}</span>
                        </div>
                    </div>
                </div>
                <div class="modal-footer">
                    <button type="button" class="btn btn-default" data-dismiss="modal">Close</button>
                    <button v-on:click.prevent="onSubmit('merchant')" type="submit" class="btn btn-primary hidden" id="btn-save-refund-products">Refund</button>
                    <!--<button v-on:click.prevent="onSubmit('commission')" type="submit" class="btn btn-primary" id="btn-save-refund-products-commission">Refund (Commission Only)</button>-->
                    <button v-on:click.prevent="onSubmit('commission')" type="submit" class="btn btn-primary" id="btn-save-refund-products-commission">Refund</button>
                </div>
            </form>
        </div>
    </div>

    <div class="modal fade" id="modal-refund-order" role="dialog" aria-labelledby="modal-refund-order-label">
        <div class="modal-dialog modal-xl" role="document">
            <form class="modal-content form-horizontal" id="form-refund-order">
                <div class="modal-header">
                    <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
                    <h4 class="modal-title" id="modal-refund-order-label">CLAWBACK</h4>
                </div>
                <div class="modal-body">
                    <div class="form-group" v-show="error.message != null">
                        <div class="col-sm-12 has-error text-center">
                            <span class="help-block">{{ error.message }}</span>
                        </div>
                    </div>
                    <div class="table-responsive">
						<table id="user-payouts" class="table table table-striped">
							<thead class="table__header table__header--bg-primary">
                            <tr class="table__row">
								<th class="table__cell">Associate ID</th>
								<th class="table__cell">Associate Name</th>
								<th class="table__cell">Commission Type</th>
								<th class="table__cell">Commission Period</th>
								<th class="table__cell">Commission Earned</th>
								<th class="table__cell">Commission Clawedback</th>
								<th class="table__cell">Date Clawedback</th>
								<th class="table__cell">Amount to Clawback</th>
								<th class="table__cell">Action</th>
                            </tr>
                            </thead>
                            <tbody class="table__body">
                            </tbody>
						</table>
					</div>
                </div>
                <div class="modal-footer">
                    <!--<button v-if="!order.is_clawback" v-on:click.prevent="onSubmitRefund('commission')" type="submit" class="btn btn-primary" id="btn-save-refund-commission">Refund (Commission Only)</button>-->
                    <!--<button v-if="!order.is_clawback" v-on:click.prevent="onSubmitRefund('commission')" type="submit" class="btn btn-primary disabled" id="btn-save-refund-commission">Clawback</button>-->
                    <button class="btn btn-default" data-dismiss="modal">Close</button>
                </div>
            </form>
        </div>
    </div>

    <!-- Move Order Modal -->
    <div class="modal fade" id="modal-move" role="dialog" aria-labelledby="myModalLabel">
        <div class="modal-dialog" role="document">
            <form class="modal-content form-horizontal" id="form-move" v-on:submit.prevent="save">
                <div class="modal-header">

                    <h4 class="modal-title" id="myModalLabel">ORDER ID {{ order.order_id }}</h4>
                </div>
                <div class="modal-body">
                    <div class="form-group">
                        <div class="col-sm-12">
                            <label for="invoice">Invoice</label>
                            <input class="form-control" id="invoice" v-model="order.invoice" readonly />
                        </div>
                    </div>
                    <div class="form-group">
                        <div class="col-sm-12">
                            <label for="transaction-date">Transaction Date</label>
                            <input class="form-control" id="transaction-date" v-model="order.transaction_date" onkeydown="return false"/>
                        </div>
                    </div>
                    <div class="form-group">
                        <div class="col-sm-12">
                            <label for="purchaser">Purchaser</label>
                            <input class="form-control" id="purchaser" v-model="displayPurchaser" readonly />
                        </div>
                    </div>
                    <div class="form-group">
                        <div class="col-sm-12">
                            <label for="sponsor">Sponsor</label>
                            <input class="form-control" id="sponsor" v-model="displaySponsor" readonly />
                        </div>
                    </div>
                    <div class="form-group">
                        <div class="col-sm-12">
                            <label for="new-purchaser-id">New Purchaser</label><br/>
                            <select2-autocomplete-member id="new-purchaser-id" :url="autocompleteUrl" v-model="order.new_purchaser_id"></select2-autocomplete-member>
                        </div>
                    </div>
                    <div class="form-group" v-show="error.message != null">
                        <div class="col-sm-12 has-error text-center">
                            <span class="help-block">{{ error.message }}</span>
                        </div>
                    </div>

                </div>
                <div class="modal-footer">
                    <button class="btn btn-default" data-dismiss="modal">Close</button>
                    <button type="submit" class="btn btn-primary" id="btn-save" >Update</button>
                </div>
            </form>
        </div>
    </div>

</div>

<script src="$commission_engine_api_url/js/app.js?v=$app_js_version"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/select2/4.0.7/js/select2.full.min.js"></script>
<script src="https://cdn.datatables.net/responsive/2.2.3/js/dataTables.responsive.min.js"></script>
<script src="https://cdn.datatables.net/responsive/2.2.3/js/responsive.bootstrap.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-datepicker/1.8.0/js/bootstrap-datepicker.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/eonasdan-bootstrap-datetimepicker/4.17.47/js/bootstrap-datetimepicker.min.js"></script>
<script src="$commission_engine_api_url/js/admin_clawback.js?v=1"></script>

EOS
1;